Update logging and errors

ltseeley opened the pull request:
Numerous improvements to comments, docstrings, log messages, and error handling

peterschwarz commented:
Actually meant to request changes there

ltseeley commented on src/engine.rs:
```diff
@@ -142,20 +146,23 @@ fn handle_update(
+        }
         Ok(Update::PeerConnected(_)) | Ok(Update::PeerDisconnected(_)) => {
-            debug!("Received PeerConnected/PeerDisconnected message");
+            info!("Received PeerConnected/PeerDisconnected message");
```
Fixed

ltseeley commented on src/node.rs:
```diff
@@ -105,7 +104,7 @@ impl PbftNode {
             PbftMessageType::ViewChange => self.handle_view_change(&msg, state)?,
             PbftMessageType::NewView => self.handle_new_view(&msg, state)?,
-            _ => warn!("Message type not implemented"),
+            _ => warn!("Received message with unknown type"),
```
Fixed

chenette commented:
The comments, docstrings, and log messages look good.

vaporos commented on src/engine.rs:
```diff
@@ -94,7 +94,7 @@ impl Engine for PbftEngine {
                 if node.check_faulty_primary_timeout_expired(state) {
                     warn!("Faulty primary timeout expired; proposing view change");
-                    handle_pbft_result(node.start_view_change(state, state.view + 1));
+                    log_any_error(node.start_view_change(state, state.view + 1));
```
These log_any_error() calls seem a bit awkward. In any case, this might be a good use for a macro.

ltseeley replied:
Could you explain what makes this a good use for a macro? Or do you have other suggestions for how to make this less awkward? The only thing I can think of at the moment would be to just call `.unwrap_or_else(|err| error!("{}", err)` in these situations.

ltseeley commented:
Removed the 2 commits that renamed `InternalError` to `GeneralError` and removed the `SerializationError` type. Added 5 commits to properly categorize and display the different error types.
